# Likes Report
## Jaylen Hicks
### September 12, 2024
**Who or what is the source of your personal Instagram data?**

The source of personal Instagram data is the user. By liking the posts and inteacting with data, users are fundamental in the data curation step. Instagram collects this data and stores it. This data can be used for hypothesis creation and further data analysis.

**Why was this data created and what purposes can the data serve?**

This data was created because it can be extremely beneficial to collect and analyze user data. Data that has been collected over time can be used to show trends. Analyzed trend information can be used to make hypotheses about someone and can push similar content or potential products that the user may be intersted in. 

**In what ways may this data be reliable?**

I think liked data can show reliable trends. From my experiences in how people use instagram, people choose to like posts to show support with their friends and family. Often, when friends and family share a post on Instagram, users will like that post to show the sender that they're engaging with the posts that they send. Users often try to comply with "THE ALGORITHM". Knowing that if they choose to like a certain post, they will be more likely to get similar content to the post that they like. Users spend lots of time interacting with tons of content to try to curate a feed dedicated to the content in which they like. 

**In what ways may this data be unreliable?**

I beleve this data may be unreliable in the conclusions that someone might draw while analyzing the data. Likes are public, so people can see what posts you liked. Knowing this, users may be selective about which posts they choose to like. Information relating to saved posts isn't shared with friends and followers, so I think that saved posts may better reflect which posts users would like to go back and revisit. There are additional features with saved posts, such as the opportunity to create organized collections that may want users to use the save funtion rather than the like function. Since liked posts are stored only in one collection, it may be harder to revisit liked posts.

In [1]:
import pandas as pd
import json

In [2]:
with open ('/Users/jaylen/Downloads/Jaylen_Likes/likes/liked_posts.json') as content:
    ig = json.load(content)

In [3]:
ig.keys()

dict_keys(['likes_media_likes'])

In [4]:
ig_df = pd.DataFrame(ig['likes_media_likes'])
ig_df

title  \
0                  nytcooking   
1             storyofthislife   
2                   dinahjane   
3                     memezar   
4                gracereiter1   
..                        ...   
943              videosbeauty   
944                   luv_jjp   
945  i.cant.choose.a.username   
946               bitchistics   
947               brentrivera   

                                      string_list_data  
0    [{'href': 'https://www.instagram.com/p/C9u_qgh...  
1    [{'href': 'https://www.instagram.com/reel/C-8E...  
2    [{'href': 'https://www.instagram.com/reel/C_Gp...  
3    [{'href': 'https://www.instagram.com/p/C-8eIEP...  
4    [{'href': 'https://www.instagram.com/reel/C9dL...  
..                                                 ...  
943  [{'href': 'https://www.instagram.com/p/BHC9XLR...  
944  [{'href': 'https://www.instagram.com/p/BG8A-ss...  
945  [{'href': 'https://www.instagram.com/p/BHBLdVF...  
946  [{'href': 'https://www.instagram.com/p/BGm17ss...  
947  [{'href': 'https://www.instagram.com/p/LzR5XUA...  

[948 rows x 2 columns]

In [5]:
ig_df = pd.DataFrame(ig['likes_media_likes'])
ig_df

title  \
0                  nytcooking   
1             storyofthislife   
2                   dinahjane   
3                     memezar   
4                gracereiter1   
..                        ...   
943              videosbeauty   
944                   luv_jjp   
945  i.cant.choose.a.username   
946               bitchistics   
947               brentrivera   

                                      string_list_data  
0    [{'href': 'https://www.instagram.com/p/C9u_qgh...  
1    [{'href': 'https://www.instagram.com/reel/C-8E...  
2    [{'href': 'https://www.instagram.com/reel/C_Gp...  
3    [{'href': 'https://www.instagram.com/p/C-8eIEP...  
4    [{'href': 'https://www.instagram.com/reel/C9dL...  
..                                                 ...  
943  [{'href': 'https://www.instagram.com/p/BHC9XLR...  
944  [{'href': 'https://www.instagram.com/p/BG8A-ss...  
945  [{'href': 'https://www.instagram.com/p/BHBLdVF...  
946  [{'href': 'https://www.instagram.com/p/BGm17ss...  
947  [{'href': 'https://www.instagram.com/p/LzR5XUA...  

[948 rows x 2 columns]

In [6]:
ig_df.groupby('title').count().\
sort_values('string_list_data', ascending = False)

string_list_data
title                                          
fifthharmony                                 10
avatarthelastairbender                        9
dinahslove97                                  7
bitch                                         7
stolen___memeznthingz                         5
...                                         ...
6.67.am                                       1
5hfanclubbrazil                               1
3dsworld                                      1
23_mun1_09                                    1
2.718281828459045235360287420                 1

[786 rows x 1 columns]

In [7]:
ig_df.shape

(948, 2)

On TikTok, Liked Videos and Favorited Videos serve similar purposes of showing content creators that they enjoyed the content presented. From my observations, typically people choose to like or favorite a post to save it in their respective feeds and to revisit their disired content at a later date. Because of this, I think thate there will be a lot more likes than Favorited Posts. I think if users are more selective about what posts they favorite, it will be easier to go back and try to find a specific post , rather than scrolling through a sea of post to find what they are looking for.

Also, liked posts are used to reccomend similar posts. Users are cognizant of the algorithm, and often will use likes to curate their For You Page. This may lead to a higher number of likes represented than Favorited Posts.

In [8]:
with open ('/Users/jaylen/Downloads/Jaylen_Tiktok_Data.json') as ttcontent:
    tt = json.load(ttcontent)

In [9]:
tt.keys()

dict_keys(['Activity', 'Ads and data', 'App Settings', 'Comment', 'Direct Messages', 'Income Plus Wallet Transactions', 'Poi Review', 'Profile', 'Tiktok Live', 'Tiktok Shopping', 'Video'])

In [10]:
ttfave_df = pd.DataFrame(tt['Activity']['Favorite Videos']['FavoriteVideoList'])
ttfave_df

Date                                               Link
0    2024-08-16 02:35:50  https://www.tiktokv.com/share/video/7398623451...
1    2024-08-12 16:36:49  https://www.tiktokv.com/share/video/7399738827...
2    2024-08-12 07:15:52  https://www.tiktokv.com/share/video/7370018709...
3    2024-08-11 18:19:29  https://www.tiktokv.com/share/video/7395601937...
4    2024-07-26 10:02:14  https://www.tiktokv.com/share/video/7387254450...
..                   ...                                                ...
302  2020-08-01 22:57:35  https://www.tiktokv.com/share/video/6855871964...
303  2020-05-23 05:24:37  https://www.tiktokv.com/share/video/6828616673...
304  2020-02-05 22:54:24  https://www.tiktokv.com/share/video/6790048842...
305  2020-01-27 03:23:33  https://www.tiktokv.com/share/video/6785708083...
306  2019-08-15 02:15:51  https://www.tiktokv.com/share/video/6725098929...

[307 rows x 2 columns]

In [11]:
ttlike_df = pd.DataFrame(tt['Activity']['Like List']['ItemFavoriteList'])
ttlike_df

Date                                               Link
0     2024-08-27 08:04:12  https://www.tiktokv.com/share/video/7403498582...
1     2024-08-27 02:05:46  https://www.tiktokv.com/share/video/7407558547...
2     2024-08-26 23:21:11  https://www.tiktokv.com/share/video/7407590779...
3     2024-08-26 21:56:07  https://www.tiktokv.com/share/video/7210860724...
4     2024-08-26 19:36:28  https://www.tiktokv.com/share/video/7407475407...
...                   ...                                                ...
5584  2022-02-07 23:55:06  https://www.tiktokv.com/share/video/7061662338...
5585  2022-02-07 22:56:26  https://www.tiktokv.com/share/video/7061376311...
5586  2022-02-07 21:54:33  https://www.tiktokv.com/share/video/7060632939...
5587  2022-02-07 15:54:39  https://www.tiktokv.com/share/video/7060660238...
5588  2022-02-07 15:45:08  https://www.tiktokv.com/share/video/7061353965...

[5589 rows x 2 columns]

In [12]:
ttlike_df.tail(10)

Date                                               Link
5579  2022-02-08 18:57:55  https://www.tiktokv.com/share/video/7060188393...
5580  2022-02-08 18:53:49  https://www.tiktokv.com/share/video/7062398160...
5581  2022-02-08 13:02:46  https://www.tiktokv.com/share/video/7061972009...
5582  2022-02-08 04:02:06  https://www.tiktokv.com/share/video/7056608623...
5583  2022-02-08 03:58:40  https://www.tiktokv.com/share/video/7061790358...
5584  2022-02-07 23:55:06  https://www.tiktokv.com/share/video/7061662338...
5585  2022-02-07 22:56:26  https://www.tiktokv.com/share/video/7061376311...
5586  2022-02-07 21:54:33  https://www.tiktokv.com/share/video/7060632939...
5587  2022-02-07 15:54:39  https://www.tiktokv.com/share/video/7060660238...
5588  2022-02-07 15:45:08  https://www.tiktokv.com/share/video/7061353965...

In [13]:
ttfave_df.tail(10)

Date                                               Link
297  2021-05-20 03:24:18  https://www.tiktokv.com/share/video/6964005843...
298  2021-05-06 07:56:44  https://www.tiktokv.com/share/video/6958242906...
299  2021-03-13 09:42:43  https://www.tiktokv.com/share/video/6932853917...
300  2020-10-05 23:23:00  https://www.tiktokv.com/share/video/6876949736...
301  2020-10-04 00:17:38  https://www.tiktokv.com/share/video/6875302335...
302  2020-08-01 22:57:35  https://www.tiktokv.com/share/video/6855871964...
303  2020-05-23 05:24:37  https://www.tiktokv.com/share/video/6828616673...
304  2020-02-05 22:54:24  https://www.tiktokv.com/share/video/6790048842...
305  2020-01-27 03:23:33  https://www.tiktokv.com/share/video/6785708083...
306  2019-08-15 02:15:51  https://www.tiktokv.com/share/video/6725098929...

Yes, I believe my hypothesis was proven correct. I am much more selective with the posts that I favorite, with there being 306 Favorited posts in comparison to the 5000+ liked posts that I have favorited. I would think that it would be more accurate to compare this with other's data to see if others share the same trends.